include/CL/opencl.hpp

```diff
@@ -988,6 +1006,167 @@ static inline cl_int errHandler (cl_int err, const char * errStr = NULL)
 //! \endcond
 
 
+#if defined(cl_khr_command_buffer)
```

Comment:
I see the `cl_ext.h` file is included in `opencl.hpp` conditionally:
```C++
#if defined(CL_HPP_USE_CL_DEVICE_FISSION) || defined(CL_HPP_USE_CL_SUB_GROUPS_KHR)
#include <CL/cl_ext.h>
#endif
```
I don't know these definitions. Is the inclusion guaranteed?

Reply:
Hrm, this seems like a bug.  Because we're including `CL/opencl.h` unconditionally, which includes `CL/cl_ext.h` unconditionally, there's no reason to conditionally include `CL/cl_ext.h`.  I'll file a PR to fix this.

This also means that EwanC's comment is correct and we can use the function pointer typedefs in `cl_ext.h` vs. duplicating them here.

Reply:
PR to fix (remove) the conditional inclusion is here: https://github.com/KhronosGroup/OpenCL-CLHPP/pull/195

Reply:
Thanks for you comments.

This redeclaration function pointers is requirement by macro: https://github.com/KhronosGroup/OpenCL-CLHPP/blob/main/include/CL/opencl.hpp#L733. It is adding prefix 'PFN_'. I used already existing impl but I am agree, it would be better to reuse the ones defined in cl_ext.h.

Could you advise me the best solution for this, shall I create a new Macro ?

Reply:
I used c++ aliases to _fn pointers from cl_ext.h.